examples/srlinux/2node-srl-ixr6-with-oc-services.pbtxt

```diff
@@ -0,0 +1,90 @@
+# This topology consists of two Nokia SR Linux IXR-6e nodes running 22.6.3 release and connected with a single link over ethernet-1/1 interfaces.
+# Each of the nodes comes up with a startup config which is sourced from 22.6.3-opeconfig.cfg.json file;
+# the startup config contains a self-signed TLS cert and has gnmi, gnoi, gribi, p4rt services enabled
+#
+# Read more about this topology at http://learn.srlinux.dev/tutorials/infrastructure/kne/srl-with-kne/
+
+name: "2-srl-ixr6"
+nodes: {
+    name: "srl1"
+    vendor: NOKIA
+    model: "ixr6e"
+    config:{
```

Comment:
I see the cert field is left out in the proto, and its entirely defined by the config file. is the intention for the user to prefer config defined certs over proto defined certs? Is the node implementation of Certer intended to be used only in case where the user wants to create/delete new certs after startup without pushing config?

Just a general question so we can clearly communicate to users what is the best practice. Right now it is not entirely straightforward. I think the problem in general is the proto has to be looked at in combination with the config because proto fields can be empty while there are actually certs on the device

Reply:
yes, the issue with self-signed certs that we have with srlinux today is that if you specify the self-signed certificated in the topo, then this cert is generated _after_ the node boots.

In order to use that certificate you have to create a binding between the TLS profile and services that intend to use it. When a certificate embedded in the config, you can provide all the bindings right away, without needing to configure them after the lab deployment.

When `srl-controller` adds `ConfigPush` you will be able to make use of the topo-defined certificates by pushing config that will make the binding between the self-generated cert and the services (gnmi, gnoi, etc) that want to use it.

Of course, you still can generate a self signed cert on srlinux and create the necessary bindings using gnmi, I just thought that packing it all in a config file is _good enough_, as far as self-signed certs are concerned.